Windows - wrong input - error

### Bug report

When writting a wrong command:
```
PS C:\Users\Bernd> kn services listr
Error: unknown sub-command 'listr' for 'kn service'. Available sub-commands: create, delete, describe, export, list, update
panic: regexp: Compile(`'(C:\Windows\system32\kn.exe\s.+?)'`): error parsing regexp: invalid escape sequence: `\k`

goroutine 1 [running]:
regexp.MustCompile(0xc0006b6cc0, 0x23, 0xc000163ec0)
        /root/.gvm/gos/go1.14.12/src/regexp/regexp.go:311 +0x159
main.extractCommandPathFromErrorMessage(0xc0000f8800, 0x74, 0xc00009e120, 0x1a, 0xc000163f00, 0x1)
        /home/prow/go/src/knative.dev/client/cmd/kn/main.go:190 +0xab
main.printError(0x1a325a0, 0xc00057f760)
        /home/prow/go/src/knative.dev/client/cmd/kn/main.go:182 +0x17a
main.main()
        /home/prow/go/src/knative.dev/client/cmd/kn/main.go:40 +0xa0
```

### Expected behavior

A more "readbale" message!

### Steps to reproduce the problem

Windows 10 

### kn version

```
kn version
Version:      v0.18.4
Build Date:   2020-11-26 07:38:41
Git Revision: 4e09c152
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.18.0)
* Eventing
  - sources.knative.dev/v1alpha2 (knative-eventing v0.18.0)
  - eventing.knative.dev/v1beta1 (knative-eventing v0.18.0)

```

thanks for the report @Berndinox !

We try to parse the sub-string `'kn service'` from the following error message for a follow up tip (`Run 'kn service --help' for usage`).
```
Error: unknown sub-command 'listr' for 'kn service'. Available sub-commands: create, delete, describe, export, list, update
```
however, we're using `os.Args[0]` to grab the `kn` string to match the pattern, which on Windows platform is expanding to its absolute path (`C:\Windows\system32\kn.exe`) and the `\k` collides with the regexp expressions. See the formed pattern below:
```
panic: regexp: Compile(`'(C:\Windows\system32\kn.exe\s.+?)'`): error parsing regexp: invalid escape sequence: `\k`
```

The solution is to simply use `"kn"` string to match the required among error message.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -17,6 +17,10 @@
 |===
 | | Description | PR
 
+| 🐛
+| Fix parsing the kn commands from error message
+| https://github.com/knative/client/pull/1174[#1174]
+
 | 🐣
 | Removes unnecessary help message introduced for machine-readable outputs for command 'kn source binding describe'
 | https://github.com/knative/client/pull/1158[#1158]

--- a/cmd/kn/main.go
+++ b/cmd/kn/main.go
@@ -179,7 +179,11 @@ func printError(err error) {
 	var runError *runError
 	if !errors.As(err, &runError) {
 		// Print help hint only if its not a runError occurred when executing a command
-		fmt.Fprintf(os.Stderr, "Run '%s --help' for usage\n", extractCommandPathFromErrorMessage(err.Error(), os.Args[0]))
+		// The error message contains pattern 'kn CMDs', thus send 'kn' string to match the pattern.
+		// Sending `os.Args[0]` instead, may result in panics while compiling the regexp, as it
+		// may expand to the absolute path of the kn binary and the path may collide with regexp expressions.
+		// see https://github.com/knative/client/issues/1172
+		fmt.Fprintf(os.Stderr, "Run '%s --help' for usage\n", extractCommandPathFromErrorMessage(err.Error(), "kn"))
 	}
 }
 

